### PR TITLE
Fix the issue of nothing to plot #1026 and big merge with branch development 

### PR DESCRIPTION
**Pull request by NDManh**

### CODE REVIEW

File: code-postprocessing/bbob_pproc/ppfigdim.py [hunk 1]
```diff
@@ -480,6 +483,13 @@ def plot(dsList, valuesOfInterest=None, styles=styles):
             res.extend(plt.plot(tmp[:, 0], maxevals / tmp[:, 0]**ynormalize_by_dimension,
                        color=styles[len(valuesOfInterest) - 1]['color'],
                        ls='', marker='x', markersize=20))
+            if no_target_reached:
+                ylim_maxevals = max(maxevals / tmp[:, 0]**ynormalize_by_dimension)
+                ylim = (ylim[0], math.ceil(ylim_maxevals))
+                plt.annotate('no target reached', xy=(0.5, 0.5), xycoords='axes fraction', fontsize=14,
+                            horizontalalignment='center', verticalalignment='center')
+            else:
+                ylim = (ylim[0], math.ceil(ylim[1]))
```
Comment (nikohansen) [inline code-postprocessing/bbob_pproc/ppfigdim.py]: looking at the code more carefully, it looks like `ppfig.discretize_limits()` is meant to do this job (later) in `beautify()`. Hence `math.ceil` is not needed here or in line 488 (and we should see the same result without).

**Comment by nikohansen** [comment]
It looks like the second commit should not be necessary. It shouldn't be too difficult to avoid getting this large change-set?

File: code-postprocessing/bbob_pproc/ppfigdim.py (hunk 1 above)
Comment (NDManh) [inline code-postprocessing/bbob_pproc/ppfigdim.py]: I have tried to change the code as you suggested, that is without using `math.ceil`:
```
if 1 < 3: # maxevals
            ylim = plt.ylim()
            res.extend(plt.plot(tmp[:, 0], maxevals / tmp[:, 0]**ynormalize_by_dimension,
                       color=styles[len(valuesOfInterest) - 1]['color'],
                       ls='', marker='x', markersize=20))
            if no_target_reached:
                ylim_maxevals = max(maxevals / tmp[:, 0]**ynormalize_by_dimension)
                ylim = (ylim[0], ylim_maxevals)
                plt.annotate('no target reached', xy=(0.5, 0.5), xycoords='axes fraction', fontsize=14,
                            horizontalalignment='center', verticalalignment='center')
            plt.ylim(ylim)
```
However, the obtained result is not the same, for example : 
- With the current code: 
[ppfigdim_f007.pdf](https://github.com/numbbo/coco/files/467257/ppfigdim_f007.pdf)
[ppfigdim_f023.pdf](https://github.com/numbbo/coco/files/467258/ppfigdim_f023.pdf)
- With the revised code (without using `math.ceil`):  
[ppfigdim_f007.pdf](https://github.com/numbbo/coco/files/467260/ppfigdim_f007.pdf)
[ppfigdim_f023.pdf](https://github.com/numbbo/coco/files/467262/ppfigdim_f023.pdf)

**Comment by nikohansen** [comment]
I would like to understand why beautify doesn't do the job before to move on.

**Comment by nikohansen** [comment]
I suggest also to first checkout the latest version of the branch and then add the small change, to un-clutter the commit (which has now 180 files changed).

**Comment by NDManh** [comment]
> I would like to understand why beautify doesn't do the job before to move on.

1) Firstly, let's talk about the command `ylim = plt.ylim()` (line 482 in `ppfigdim.py`). This command gives different values to `ylim` in different runs of the post-processing on the same data set. For instance, I've tried to test on the data of the function `f23`, and obtained: 
`ylim = (0.0, 1.2000000000000002)` or `ylim = (0.0, 0.40000000000000002)` or `ylim = (0.0, 0.70000000000000007)`.

2) While in the case `ylim = (0.0, 1.2000000000000002)`, the result is the same for the code with or without using `math.ceil`; the difference appears for the other cases.

With the current code (using `math.ceil`), due to the command `ylim = (ylim[0], math.ceil(ylim[1]))` , the value `ylim = (0.0, 0.40000000000000002)` or `ylim = (0.0, 0.70000000000000007)` becomes `ylim = (0.0, 1.0)`, then `ppfig.discretize_limits()` in `beautify()` transforms it to `(ymin_new, ymax_new) = (0.6309573444801932, 3.000003)`. That is why we see the figure like  [ppfigdim_f023.pdf](https://github.com/numbbo/coco/files/475007/ppfigdim_f023.pdf)

In the case we do not use `math.ceil`, then `ppfig.discretize_limits()` in beautify() will transform `ylim = (0.0, 0.70000000000000007)` to `(ymin_new, ymax_new) = (0.6309573444801932, 1.0000009999999999)`, thus the figure looks like [ppfigdim_f023.pdf](https://github.com/numbbo/coco/files/475041/ppfigdim_f023.pdf)

**Comment by nikohansen** [comment]
OK, thanks! Without `ceil`. If we don't like the result without `ceil`, we should rather change `discretize_limits`.

**Comment by NDManh** [comment]
In fact, we do not adapt `max-evals` in this case, that is there is something rather than nothing (you can see the figure in the case using `math.ceil`). But it seems to be hidden, maybe it is because the `ylim` is too small?

**Pull request closed without merging**